icu4c/source/common/uresdata.h

```diff
@@ -512,20 +514,27 @@ class ResourceDataValue : public ResourceValue {
 public:
     ResourceDataValue() :
         pResData(nullptr),
+        validLocaleDataEntry(nullptr),
         res(static_cast<Resource>(URES_NONE)),
         fTraceInfo() {}
     virtual ~ResourceDataValue();
 
     void setData(const ResourceData &data) {
         pResData = &data;
     }
+    
+    void setValidLocaleDataEntry(UResourceDataEntry *entry) {
+        validLocaleDataEntry = entry;
+    }
 
     void setResource(Resource r, ResourceTracer&& traceInfo) {
         res = r;
         fTraceInfo = traceInfo;
     }
 
     const ResourceData &getData() const { return *pResData; }
+    UResourceDataEntry *getValidLocaleDataEntry() const { return validLocaleDataEntry; }
+    Resource getRes() const { return res; }
```

Comment:
```suggestion
    Resource getResource() const { return res; }
```

Let's make this a proper pair with setResource().